paws_collector.js

```diff
@@ -440,6 +440,75 @@ class PawsCollector extends AlAwsCollector {
         this.reportDDMetric('api_throttling', 1)
         return cloudwatch.putMetricData(params, callback);
     };
+
+    /**
+     * Report the client errors and show case on DDMetrics and cloudwatch
+     * @param callback 
+     * @param error 
+     */
+    reportClientError(error, callback) {
```

Comment:
Just so I'm clear on work left to be done here. Is the intention for subclasses of paws-collector to call these methods? 

If so I don't think that's the right approach. I wouldn't want this behaviour to be done at the subclass level, IMO we'd want to call this function [here](https://github.com/alertlogic/paws-collector/blob/bc67a70c8463e9c05037a54982a000c3d9dd69a2/paws_collector.js#L380-L382):

```javascript
function(asyncCallback) {
    return collector.pawsGetLogs(pawsState.priv_collector_state, asyncCallback);
}
```

@JDragovichAlertLogic  @kkuzmin thoughts?

Reply:
I think that because each collector si going to have a different schema for errors (ie. they might be formmated by an SDK) we will need each collector to do its own logic ion formatting errors.

I don't know if there is going to be a library level solution fo this issue. For example, in the o365 collector, we're pulling from two different apis in sequence through an SDK.

We could call this method in the same way we call the throttling metric method.

https://github.com/alertlogic/paws-collector/blob/bc67a70c8463e9c05037a54982a000c3d9dd69a2/collectors/okta/okta_collector.js#L79-L85

I mean we could handle this on the base class level. but we would need to force each collector to implement a `handleClientError` method because i dont really thing we can trust thee sdks to give us a raw nodejs error.

So a reasonable example is that we have funky error codes coming back from the okta api that we might want to capture that give us more info than just the http error. Not sure how the sdk handles this.

https://developer.okta.com/docs/reference/error-codes/

Reply:
Ok after some kibitzing offline, Dan and I hashed out  plan for how this can be implemented.

Dan is correct that we should define a reasonable shape for this error object, and it should probably resemble the standard nodejs error.

However we should call this reporting method after the logs are collected and rely on the `asyncCallback` in `pawsGetLogs` to pass the error in. Each collector will need to make sure hte error object is the correct shape.

somthing like this:

``` javascript
function(asyncCallback) {
        return collector.pawsGetLogs(pawsState.priv_collector_state, (err, ...remainingParams) => {
      if (err) {
        this.reportClientError(err);
        asyncCallback(err);
      } else {
        this.reportClientOk();
        asyncCallback(null, ...remainingParams)
      }
    });
},
```